rancher2/resource_rancher2_project.go

```diff
@@ -108,12 +108,15 @@ func resourceRancher2ProjectCreate(d *schema.ResourceData, meta interface{}) err
 				if IsConflict(err) || IsForbidden(err) {
 					return resource.RetryableError(err)
 				}
-				return resource.NonRetryableError(err)
+				// Checking error due to ActionSetpodsecuritypolicytemplate() issue
+				if error.Error(err) != "unexpected end of JSON input" {
+					return resource.NonRetryableError(err)
+				}
 			}
 			return nil
 		})
 		if err != nil {
-			return err
+			return fmt.Errorf("[ERROR] waiting for pod_security_policy_template_id (%s) to be set on project (%s): %s", pspID, newProject.ID, err)
```

Comment:
Would generally prefer `%w` used for wrapped errors but then I see it's not used throughout the file so should leave it "as is". https://stackoverflow.com/a/61287626
It's also weird that error is prefixed with `[ERROR]` which doesn't seem to make sense (it makes sense for `log.Printf`) but again it's an established practice in this file - so going for consistency.